Bonestorm RP Fix

TheAsaSpades opened the pull request:
Bonestorm was only having its base runic power cost (10) added to Insatiable Hunger, this fix adds the correct amount of runic power spent on Bonestorm to Insatiable Hunger

TheAsaSpades committed:
Add the correct amount of runic power spent on Bonestorm to Insatiable Hunger

TheAsaSpades committed:
Insatiable Hunger rp fix, apl update

Turns out every spell being added for Blood to Insatiable Hunger was using the wrong rp. Fixed it so the correct rp was added. Also updated the apl to use bonestorm, and to pool rp for swarming mists now and use it inside drw when venthyr.

nyterage commented on engine/class_modules/apl/apl_death_knight.cpp:
```diff
@@ -108,32 +108,33 @@ void blood( player_t* p )
   default_->add_action( "blooddrinker,if=!buff.dancing_rune_weapon.up&(!covenant.night_fae|buff.deaths_due.remains>7)" );
   default_->add_action( "blood_boil,if=charges>=2&(covenant.kyrian|buff.dancing_rune_weapon.up)" );
-  default_->add_action( "raise_dead" );
+//  default_->add_action( "raise_dead" );
```
Why remove raise dead here? its not on a GCD, and is just a free dps increase? Doesn't make much sense to remove it.

TheAsaSpades replied:
Cause it's used 2 lines before, and you want to use it before blooddrinker so you don't interrupt blooddrinker.

Armadk commented on engine/class_modules/sc_death_knight.cpp:
```diff
@@ -7793,7 +7793,7 @@ double death_knight_t::resource_loss( resource_e resource_type, double amount, g
 
     // If an action is linked, fetch its base cost. Exclude Bonestorm from this otherwise it uses the base cost for Insatiable Hunger instead of the actual rp spent.
-    if ( action && action->id != 194844 )
+    if ( action && specialization() != DEATH_KNIGHT_BLOOD )
```
Stepping through the RP spending spells, it looks like only Bonestorm, and Death Strike do this.  

I tried with death coil while wearing deadliest coil, and I was getting the base cost, rather than the actual cost.  Interestingly,  when I first implemented this death strike was giving the base cost, rather than actual.  I don't have data for bonestorm in my notes from that time.

Armadk replied:
Re-tested this today after reporting the issue with Death Strike, and it now correctly grants a value of 90 (as if 45 rp was spent) when ossuary is up.  Only Bonestorm should need the adjustment.  Bonestorm on current build gives 200 value (100 rp) with a full rp bar.

TheAsaSpades committed:
Revert "Insatiable Hunger rp fix, apl update"

This reverts commit 22da8f131bc7fcec1471e661d1e1810d6b379346.

TheAsaSpades committed:
[Death Knight] RP fix for Blood for Insatiable Hunger

The actual rp spent on Bonestorm and Death Strike was not being added to Insatiable Hunger calculations. It was always adding the base cost, so 10 for Bonestorm, even if more was spent, and 45 for Death Strike, even with Ossuary up.

TheAsaSpades committed:
Merge branch 'simulationcraft:shadowlands' into shadowlands_dk_bonestorm_insatiable_hunger_fix

Armadk commented on engine/class_modules/sc_death_knight.cpp:
```diff
@@ -7792,10 +7792,12 @@ double death_knight_t::resource_loss( resource_e resource_type, double amount, g
+    if ( action && specialization() != DEATH_KNIGHT_BLOOD )
       base_rp_cost = action -> base_costs[ RESOURCE_RUNIC_POWER ];
 
+
```
```suggestion

```

TheAsaSpades committed:
Update engine/class_modules/sc_death_knight.cpp

Co-authored-by: Arma <adam.shewchuk@gmail.com>

Armadk committed:
Death Strike no longer needs special handling.  It was fixed in game.

Armadk committed:
Adjust comment
